### PR TITLE
remove APIGen project

### LINKED ISSUE #742: Re-consider per-build AMQP API generation
See [this comment](https://github.com/rabbitmq/rabbitmq-dotnet-client/issues/741#issuecomment-593123031) in #741

### PR DESCRIPTION
**Pull request by bollhals**
## Proposed Changes

Removes the ApiGen project and brings code into separate files (resolves #742)

## Types of Changes

What types of changes does your code introduce to this project?
_Put an `x` in the boxes that apply_

- [x] Bug fix (non-breaking change which fixes issue #742)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause an observable behavior change in existing systems)
- [ ] Documentation improvements (corrections, new content, etc)
- [x] Cosmetic change (whitespace, formatting, etc)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating
the PR. If you're unsure about any of them, don't hesitate to ask on the
mailing list. We're here to help! This is simply a reminder of what we are
going to look for before merging your code._

- [x] I have read the `CONTRIBUTING.md` document
- [x] I have signed the CA (see https://cla.pivotal.io/sign/rabbitmq)
- [ ] All tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in related repositories

### CODE REVIEW

**Comment by bollhals** [comment]
I have created a new PR for this, as it was too difficult to update the old one. 

There are more changes coming for formatting and stuff (similar to original PR #877), which I'll add as separate commits.

File: projects/RabbitMQ.Client/client/framing/BasicAck.cs [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+//
+//  Copyright (c) 2007-2020 VMware, Inc.  All rights reserved.
+//---------------------------------------------------------------------------
+
+using System.Text;
+
+namespace RabbitMQ.Client.Framing.Impl
+{
+    /// <summary>Autogenerated type. Private implementation class - do not use directly.</summary>
```
Comment (michaelklishin) [inline projects/RabbitMQ.Client/client/framing/BasicAck.cs:36]: We should remove these "Autogenerated type" mentions.

Comment (bollhals) [reply projects/RabbitMQ.Client/client/framing/BasicAck.cs:36]: Saw them as well, in my local changes I already started to remove them.

**Comment by michaelklishin** [comment]
@bording @danielmarbach @stebet @lukebakken @kjnilsson before I pull the trigger on this, any last minute objections to removing the API generator? The protocol doesn't change often but we do add things to it occasionally. I personally think it'd be a great simplification to the build system and the codebase, so we should do it, and maybe other clients will follow suit eventually. But maybe it's just me :)

**Comment by danielmarbach** [comment]
I agree it can go. Makes it also simpler to understand and evolve if needed

**Comment by stebet** [comment]
I agree as well. Makes it simpler to build and make beneficial changes.

**Comment by bollhals** [comment]
@reviewers, look at the actual changes made in the commits after the first one. The first is the removal of the project (split of the generated code into their own classes, and removal of the old generator project)

**Comment by bollhals** [comment]
Ok I think this is done from my side. This can be reviewed.

One thing that I saw was some simplifications around method / class id. I'll check this out in a separate PR.

File: build.bat [hunk 2]
```diff
@@ -1,5 +1,4 @@
 @ECHO OFF
 set DOTNET_CLI_TELEMETRY_OPTOUT=1
 dotnet restore .\RabbitMQDotNetClient.sln
-dotnet run -p .\projects\Apigen\Apigen.csproj --apiName:AMQP_0_9_1 .\projects\specs\amqp0-9-1.stripped.xml .\gensrc\autogenerated-api-0-9-1.cs
```
Comment (bollhals) [inline build.bat:4]: Build is still failing. Am I missing something or does this not have the desired effect?

**Comment by lukebakken** [comment]
@michaelklishin I just fixed up concourse to take this change into account. Thanks for reviewing and merging.

https://github.com/rabbitmq/rabbitmq-ci/commit/2d3e9680c2aafb79446437e486b7b2ad491c5901

File: build.bat (hunk 2 above)
Comment (lukebakken) [reply build.bat:4]: I had to fix some scripts - https://github.com/rabbitmq/rabbitmq-dotnet-client/pull/924#issuecomment-676561180

**Pull request merged**